.github/workflows/build_documentation.yml

```diff
@@ -40,14 +43,33 @@ jobs:
           name: nix-community
           authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
 
-      - run: |
+      - name: Build docs
+        run: |
           set -ex
-          mkdir -p docs-build
-          nix build github:nix-community/nixvim#docs
-          cp -r result/share/doc/nixvim/* docs-build
-          nix build github:nix-community/nixvim/nixos-24.05#docs
-          # TODO: use the new path when 24.11 hits
-          cp -r result/share/doc docs-build/stable
+
+          # 1: branch
+          # 2: dest-dir (relative to /nixvim/)
+          build() {
+            flake="github:${repo}${1:+/$1}"
+            baseHref="/${repoName}/${2:+$2/}"
+            destDir="${out}${baseHref}"
+
+            # Build docs for the given flake ref, overriding baseHref in the derivation args
+            nix build --impure --expr '(builtins.getFlake "'"$flake"'").outputs.packages.${builtins.currentSystem}.docs.override {
+              baseHref = "'"$baseHref"'";
+            }'
```

Comment:
Note: overriding `baseHref` relies on the 24.05 branch _having_ a `baseHref` argument. This is added by #2538

Reply:
(Yes it's messy that we need to use `getFlake` and `currentSystem` in an expr, but unfortunately `nix build` can't take both a flakeref _and_ an apply function in the same way `nix eval` can) :cry: